src/basic/pidfd-util.c

```diff
@@ -19,14 +19,16 @@
 
 static int have_pidfs = -1;
 
-static int pidfd_check_pidfs(int pid_fd) {
-
-        /* NB: the passed fd *must* be acquired via pidfd_open(), i.e. must be a true pidfd! */
+static int pidfd_check_pidfs(void) {
```

Comment:
why not keep the call as it is right now, just add one twist: if the fd is passed as < 0 we'll create a pidfd locally which we use. That way we can call it either way: if we definitely have a valid pidfd, we can just pass that in, if we don't we pass -EBADF in, and all is good.

Afaics we'll continue to have one call site where that would be beneficial, no?